apps/api/src/lib/id-generation/id-management.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { v4 as uuidv4 } from 'uuid';
+import { IIdManager } from '../../domain/interfaces/id-manager.interface';
+import { AggregateId } from '../../domain/types/AggregateId';
+import { InternalError } from '../errors/InternalError';
+import { NotAvailable } from '../types/not-available';
+import { isNotFound, NotFound } from '../types/not-found';
+import { isOK, OK } from '../types/ok';
+import { IIdRepository } from './interfaces/id-repository.interface';
+
+@Injectable()
+export class IdManagementService implements IIdManager {
```

Comment:
Rename this to `ArangoIdManagementService` to avoid confusion.

Reply:
On second examination, this is generic. You can inject any `IIdRepository` ('ID_REPOSITORY' token) and it will work. The repository implementation will be Arango specific, but you can configure it to be something else if you choose.

Reply:
We do need to decide where exactly this fits in. Is it an external lib?